pinot-core/src/main/java/org/apache/pinot/core/data/manager/offline/DimensionTableDataManager.java

```diff
@@ -87,6 +87,7 @@ public static DimensionTableDataManager getInstanceByTableName(String tableNameW
   private final AtomicInteger _loadToken = new AtomicInteger();
 
   private boolean _disablePreload = false;
+  private boolean _disallowDuplicatePrimaryKey = false;
```

Comment:
Still suggest naming it `_errorOnDuplicatePrimaryKey`. We shouldn't allow duplicate primary key in any case since it will result in inconsistent behavior.

Reply:
got it.